virtualizarr/zarr.py

```diff
@@ -134,7 +134,7 @@ def to_kerchunk_json(self) -> str:
     def replace(
         self,
         chunks: Optional[tuple[int, ...]] = None,
-        compressor: Optional[dict] = None,
+        compressor: Optional[dict] = None,  # type: ignore[valid-type]
```

Comment:
working around mypy precommit complaining: `error: Function "virtualizarr.zarr.ZArray.dict" is not valid as a type  [valid-type]`

Reply:
`virtualizarr.zarr.ZArray.dict` is a method not an attribute, so maybe mypy is complaining about something legitimate here?